### PR TITLE
Add known issue for Elastic Agent runtime state issue

### PR DESCRIPTION
**Pull request by pebrc**
Document the issue with Elastic Agent runtime state and adjust our Elastic Agent recipes to follow the advice given in the docs.

### CODE REVIEW

File: docs/release-notes/highlights-1.4.0.asciidoc [hunk 1]
```diff
@@ -30,3 +30,28 @@ kubectl patch validatingwebhookconfiguration "$WEBHOOK" \
     --patch='{"webhooks": [{"name": "elastic-es-validation-v1.k8s.elastic.co", "matchPolicy": "Exact"}, {"name": "elastic-es-validation-v1beta1.k8s.elastic.co", "matchPolicy": "Exact"}]}'
 ----
 
+* Elastic Agent currently writes its runtime state into the filesystem of its container. A consequence of this behaviour is that the identities of the Elastic Agents change on container restarts and any internal state of applications run by Elastic Agent is lost.  A workaround exists in mounting the agent-data `hostPath` volume into the Elastic Agent container in the location where the process writes its runtime state. You also have to run the Elastic Agent as the root user in order to be able to access the `hostPath` volume, as shown in the following abbreviated example:
```
Comment (alaudazzi) [inline docs/release-notes/highlights-1.4.0.asciidoc:33]:
```suggestion
* Elastic Agent currently writes its runtime state into the filesystem of its container. As a consequence, the identities of the Elastic Agents change on container restarts and any internal state of applications run by Elastic Agent is lost. As a workaround, you can mount the agent-data `hostPath` volume into the Elastic Agent container in the location where the process writes its runtime state. You also have to run the Elastic Agent as the root user to be able to access the `hostPath` volume, as shown in the following example:
```

File: docs/release-notes/highlights-1.4.0.asciidoc [hunk 2]
```diff
@@ -30,3 +30,28 @@ kubectl patch validatingwebhookconfiguration "$WEBHOOK" \
...
+      spec:
+        containers:
+        - name: agent
+          securityContext:
+            runAsUser: 0
+          volumeMounts:
+          - name: agent-data
+            mountPath: /usr/share/elastic-agent/data/elastic-agent-9b2fec/run
+----
+The `mountPath` differs from version to version as it contains the hash of the version control system reference which was used to build Elastic Agent. You can find out which path to use by either inspecting the Docker image or by running a command against the container like so:
```
Comment (alaudazzi) [inline docs/release-notes/highlights-1.4.0.asciidoc:53]:
```suggestion
The `mountPath` differs from version to version as it contains the hash of the version control system reference which was used to build Elastic Agent. You can find out which path to use by either inspecting the Docker image or by running a command against the container, as shown below:
```

File: docs/release-notes/highlights-1.4.0.asciidoc [hunk 3]
```diff
@@ -18,7 +18,7 @@ link:https://www.elastic.co/guide/en/fleet/current/elastic-agent-installation-co
 [id="{p}-140-known-issues"]
 === Known issues
 
-On Kubernetes versions 1.16 or higher, if the operator is installed using Helm and if the validating webhook is enabled, you may be prevented from <<{p}-volume-claim-templates,increasing the storage size>> of Elasticsearch `volumeClaimTemplates` even if the underlying storage class allows expansion. This is due to a change in how `admissionregistration.k8s.io/v1` resources match update requests to validating webhook endpoints. You can work around this problem by patching the validating webhook as follows:
+* On Kubernetes versions 1.16 or higher, if the operator is installed using Helm and if the validating webhook is enabled, you may be prevented from <<{p}-volume-claim-templates,increasing the storage size>> of Elasticsearch `volumeClaimTemplates` even if the underlying storage class allows expansion. This is due to a change in how `admissionregistration.k8s.io/v1` resources match update requests to validating webhook endpoints. You can work around this problem by patching the validating webhook as follows:
```
Comment (alaudazzi) [inline docs/release-notes/highlights-1.4.0.asciidoc:21]:
```suggestion
* On Kubernetes version 1.16 or higher, if the operator is installed using Helm and if the validating webhook is enabled, you might be prevented from <<{p}-volume-claim-templates,increasing the storage size>> of Elasticsearch `volumeClaimTemplates` even if the underlying storage class allows expansion. This is due to a change in how `admissionregistration.k8s.io/v1` resources match update requests to validating webhook endpoints. As a workaround, you can patch the validating webhook as follows:
```

**Review by alaudazzi — APPROVED** [review]
A few editing suggestions, otherwise LGTM.

File: docs/release-notes/highlights-1.4.0.asciidoc [hunk 4]
```diff
@@ -30,3 +30,28 @@ kubectl patch validatingwebhookconfiguration "$WEBHOOK" \
     --patch='{"webhooks": [{"name": "elastic-es-validation-v1.k8s.elastic.co", "matchPolicy": "Exact"}, {"name": "elastic-es-validation-v1beta1.k8s.elastic.co", "matchPolicy": "Exact"}]}'
 ----
 
+* Elastic Agent currently writes its runtime state into the filesystem of its container. As a consequence, the identities of the Elastic Agents change on container restarts and any internal state of applications run by Elastic Agent is lost. As a workaround, you can mount the agent-data `hostPath` volume into the Elastic Agent container in the location where the process writes its runtime state. You also have to run the Elastic Agent as the root user to be able to access the `hostPath` volume, as shown in the following example:
```
Comment (charith-elastic) [inline docs/release-notes/highlights-1.4.0.asciidoc:33]:
```suggestion
* Elastic Agent currently writes its runtime state into the filesystem of its container. As a consequence, the identity of the Elastic Agent changes on container restarts and any internal state of applications run by that Elastic Agent is lost. As a workaround, you can mount the agent-data `hostPath` volume into the Elastic Agent container in the location where the process writes its runtime state. You also have to run the Elastic Agent as the root user to be able to access the `hostPath` volume, as shown in the following example:
```

**Review by charith-elastic — APPROVED** [review]

**Comment by pebrc** [comment]
Not sure what is going on with the build `[2021-03-04T09:09:09.258Z] /go/src/github.com/elastic/cloud-on-k8s/hack/config-extractor/../manifest-gen/manifest-gen.sh: line 71: ./manifest-gen: No such file or directory
` 😕

**Comment by pebrc** [comment]
Jenkins test this please

**Pull request merged**